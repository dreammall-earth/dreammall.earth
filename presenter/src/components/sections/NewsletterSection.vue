<template>
  <div class="newsletter-section pt-4 v-container">
    <v-row class="section-header">
      <v-col>
        <h2 class="section-headline">
          {{ $t('home.newsletterSection.headline') }}
        </h2>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <h3 class="section-subheadline">
          {{ $t('home.newsletterSection.subHeadline') }}
        </h3>
        <div class="mt-8">
          <NewsletterForm ref="newsletterForm" />
        </div>
      </v-col>
    </v-row>
    <!-- <v-row class="mt-12" align="center">
      <v-col cols="5">
        <h3 class="section-subheadline">
          {{ $t('home.newsletterSection.currentNewsletter') }}
        </h3>
      </v-col>
      <v-col cols="7">
        <MainButton
          variant="download"
          size="auto"
          :label="$t('home.newsletterSection.newsletterDownloadBtn')"
          type="submit"
          download
          :href="newsletterDownloadPath"
        >
        </MainButton>
      </v-col>
    </v-row> -->
  </div>
</template>

<script lang="ts" setup>
// import MainButton from '#components/buttons/MainButton.vue'
import { ref } from 'vue'

import NewsletterForm from '#components/forms/NewsletterForm.vue'

// TODO insert link to newest newsletter
// const newsletterDownloadPath = '#'
const newsletterForm = ref()

const focusInput = () => {
  if (newsletterForm.value) {
    newsletterForm.value.focusInput()
  }
}

defineExpose({
  newsletterForm,
  focusInput,
})
</script>

<style scoped lang="scss">
@import '#root/src/assets/scss/style';

.newsletter-section {
  max-width: $section-max-width-x-small;

  .section-header {
    text-align: center;

    h2.section-headline {
      @include section-content-headline-font;

      background: linear-gradient(
        255deg,
        $font-color-default 23.75%,
        $font-color-headline-transition-end 66.34%
      );
      -webkit-background-clip: text; /* stylelint-disable-line */
      background-clip: text;
      -webkit-text-fill-color: transparent;
    }
  }

  h3.section-subheadline {
    font-family: $font-family-default;
    font-size: 1.25rem;
    font-style: normal;
    font-weight: 300;
    line-height: normal;
    color: $font-color-default;
  }
}
</style>
