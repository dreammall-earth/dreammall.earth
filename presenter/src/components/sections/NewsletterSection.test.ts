import { VueWrapper, mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'
import { ComponentPublicInstance } from 'vue'

import NewsletterSection from './NewsletterSection.vue'

describe('NewsletterSection', () => {
  let wrapper: VueWrapper<unknown, ComponentPublicInstance<unknown, Omit<unknown, never>>>
  const Wrapper = () => {
    return mount(NewsletterSection)
  }

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.find('.newsletter-section').exists()).toBeTruthy()
  })
})
