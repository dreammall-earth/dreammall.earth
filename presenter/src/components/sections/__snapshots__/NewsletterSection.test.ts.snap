// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewsletterSection > renders 1`] = `
<div
  class="newsletter-section pt-4 v-container"
  data-v-7fdfb879=""
>
  <div
    class="v-row section-header"
    data-v-7fdfb879=""
  >
    <div
      class="v-col"
      data-v-7fdfb879=""
    >
      <h2
        class="section-headline"
        data-v-7fdfb879=""
      >
        $t('home.newsletterSection.headline')
      </h2>
    </div>
  </div>
  <div
    class="v-row"
    data-v-7fdfb879=""
  >
    <div
      class="v-col"
      data-v-7fdfb879=""
    >
      <h3
        class="section-subheadline"
        data-v-7fdfb879=""
      >
        $t('home.newsletterSection.subHeadline')
      </h3>
      <div
        class="mt-8"
        data-v-7fdfb879=""
      >
        <div
          class="v-sheet v-theme--light bg-transparent"
          data-v-7fdfb879=""
          data-v-81cc7c1c=""
        >
          <form
            class="v-form newsletter-form"
            data-v-81cc7c1c=""
            novalidate=""
          >
            
            <div
              class="v-row"
              data-v-81cc7c1c=""
            >
              <div
                class="v-col-sm-6 v-col-12"
                data-v-81cc7c1c=""
              >
                <div
                  class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field newsletter-text"
                  data-v-81cc7c1c=""
                >
                  <!---->
                  <div
                    class="v-input__control"
                  >
                    
                    <div
                      class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
                      style="background-color: rgba(174, 179, 189, 0.50);"
                    >
                      <div
                        class="v-field__overlay"
                      />
                      <div
                        class="v-field__loader"
                      >
                        <div
                          aria-hidden="true"
                          aria-valuemax="100"
                          aria-valuemin="0"
                          class="v-progress-linear v-theme--light v-locale--is-ltr"
                          role="progressbar"
                          style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);"
                        >
                          <!---->
                          <div
                            class="v-progress-linear__background"
                            style="background-color: #3D4753; color: #fff; caret-color: #fff; width: 100%;"
                          />
                          <transition-stub
                            appear="false"
                            css="true"
                            name="fade-transition"
                            persisted="false"
                          >
                            <div
                              class="v-progress-linear__indeterminate"
                            >
                              
                              <div
                                class="v-progress-linear__indeterminate long"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              <div
                                class="v-progress-linear__indeterminate short"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              
                            </div>
                          </transition-stub>
                          <!---->
                        </div>
                      </div>
                      <!---->
                      <div
                        class="v-field__field"
                        data-no-activator=""
                      >
                        <label
                          aria-hidden="true"
                          class="v-label v-field-label v-field-label--floating"
                          for="input-0"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.firstname')
                          
                        </label>
                        <label
                          class="v-label v-field-label"
                          for="input-0"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.firstname')
                          
                        </label>
                        
                        
                        <!---->
                        <input
                          aria-describedby="input-0-messages"
                          class="v-field__input"
                          id="input-0"
                          name="firstname"
                          required=""
                          size="1"
                          type="text"
                        />
                        <!---->
                        
                        
                      </div>
                      <!---->
                      <!---->
                      <div
                        class="v-field__outline"
                      >
                        <!---->
                        <!---->
                      </div>
                    </div>
                    
                  </div>
                  <!---->
                  <!---->
                </div>
              </div>
              <div
                class="v-col-sm-6 v-col-12"
                data-v-81cc7c1c=""
              >
                <div
                  class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field newsletter-text"
                  data-v-81cc7c1c=""
                >
                  <!---->
                  <div
                    class="v-input__control"
                  >
                    
                    <div
                      class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
                      style="background-color: rgba(174, 179, 189, 0.50);"
                    >
                      <div
                        class="v-field__overlay"
                      />
                      <div
                        class="v-field__loader"
                      >
                        <div
                          aria-hidden="true"
                          aria-valuemax="100"
                          aria-valuemin="0"
                          class="v-progress-linear v-theme--light v-locale--is-ltr"
                          role="progressbar"
                          style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);"
                        >
                          <!---->
                          <div
                            class="v-progress-linear__background"
                            style="background-color: #3D4753; color: #fff; caret-color: #fff; width: 100%;"
                          />
                          <transition-stub
                            appear="false"
                            css="true"
                            name="fade-transition"
                            persisted="false"
                          >
                            <div
                              class="v-progress-linear__indeterminate"
                            >
                              
                              <div
                                class="v-progress-linear__indeterminate long"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              <div
                                class="v-progress-linear__indeterminate short"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              
                            </div>
                          </transition-stub>
                          <!---->
                        </div>
                      </div>
                      <!---->
                      <div
                        class="v-field__field"
                        data-no-activator=""
                      >
                        <label
                          aria-hidden="true"
                          class="v-label v-field-label v-field-label--floating"
                          for="input-2"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.lastname')
                          
                        </label>
                        <label
                          class="v-label v-field-label"
                          for="input-2"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.lastname')
                          
                        </label>
                        
                        
                        <!---->
                        <input
                          aria-describedby="input-2-messages"
                          class="v-field__input"
                          id="input-2"
                          name="lastname"
                          required=""
                          size="1"
                          type="text"
                        />
                        <!---->
                        
                        
                      </div>
                      <!---->
                      <!---->
                      <div
                        class="v-field__outline"
                      >
                        <!---->
                        <!---->
                      </div>
                    </div>
                    
                  </div>
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <div
              class="v-row"
              data-v-81cc7c1c=""
            >
              <div
                class="v-col-sm-6 v-col-12"
                data-v-81cc7c1c=""
              >
                <div
                  class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field newsletter-text"
                  data-v-81cc7c1c=""
                >
                  <!---->
                  <div
                    class="v-input__control"
                  >
                    
                    <div
                      class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
                      style="background-color: rgba(174, 179, 189, 0.50);"
                    >
                      <div
                        class="v-field__overlay"
                      />
                      <div
                        class="v-field__loader"
                      >
                        <div
                          aria-hidden="true"
                          aria-valuemax="100"
                          aria-valuemin="0"
                          class="v-progress-linear v-theme--light v-locale--is-ltr"
                          role="progressbar"
                          style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);"
                        >
                          <!---->
                          <div
                            class="v-progress-linear__background"
                            style="background-color: #3D4753; color: #fff; caret-color: #fff; width: 100%;"
                          />
                          <transition-stub
                            appear="false"
                            css="true"
                            name="fade-transition"
                            persisted="false"
                          >
                            <div
                              class="v-progress-linear__indeterminate"
                            >
                              
                              <div
                                class="v-progress-linear__indeterminate long"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              <div
                                class="v-progress-linear__indeterminate short"
                                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
                              />
                              
                            </div>
                          </transition-stub>
                          <!---->
                        </div>
                      </div>
                      <!---->
                      <div
                        class="v-field__field"
                        data-no-activator=""
                      >
                        <label
                          aria-hidden="true"
                          class="v-label v-field-label v-field-label--floating"
                          for="input-4"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.email')
                          
                        </label>
                        <label
                          class="v-label v-field-label"
                          for="input-4"
                        >
                          <!---->
                          
                          $t('home.newsletterSection.newsletterForm.email')
                          
                        </label>
                        
                        
                        <!---->
                        <input
                          aria-describedby="input-4-messages"
                          class="v-field__input"
                          id="input-4"
                          name="email"
                          required=""
                          size="1"
                          type="email"
                        />
                        <!---->
                        
                        
                      </div>
                      <!---->
                      <!---->
                      <div
                        class="v-field__outline"
                      >
                        <!---->
                        <!---->
                      </div>
                    </div>
                    
                  </div>
                  <!---->
                  <!---->
                </div>
              </div>
              <div
                class="v-col-sm-6 d-flex align-center v-col-12"
                data-v-81cc7c1c=""
              >
                <button
                  class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--variant-elevated main-button main-button--form-submit main-button-auto w-100"
                  data-v-81cc7c1c=""
                  data-v-9518dada=""
                  style="background-color: #23ad5b; color: #fff; caret-color: #fff; height: auto; width: auto;"
                  type="submit"
                >
                  
                  <span
                    class="v-btn__overlay"
                  />
                  <span
                    class="v-btn__underlay"
                  />
                  
                  <!---->
                  <span
                    class="v-btn__content"
                    data-no-activator=""
                  >
                    
                    
                    <!--v-if-->
                    <span
                      class="main-button-content"
                      data-v-9518dada=""
                    >
                      $t('home.newsletterSection.newsletterForm.submitBtn') 
                      <!--v-if-->
                    </span>
                    
                    
                  </span>
                  <!---->
                  <!---->
                </button>
              </div>
            </div>
            <div
              class="v-row"
              data-v-81cc7c1c=""
            >
              <div
                class="v-col v-col-12 d-flex align-start"
                data-v-81cc7c1c=""
              >
                <div
                  class="v-input v-input--horizontal v-input--density-compact v-locale--is-ltr v-input--dirty v-checkbox"
                  data-v-81cc7c1c=""
                >
                  <!---->
                  <div
                    class="v-input__control"
                  >
                    
                    <div
                      class="v-selection-control v-selection-control--density-compact v-checkbox-btn"
                    >
                      <div
                        class="v-selection-control__wrapper"
                      >
                        <!---->
                        <div
                          class="v-selection-control__input"
                        >
                          
                          <i
                            aria-hidden="true"
                            class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default"
                          />
                          <input
                            aria-describedby="checkbox-7-messages"
                            aria-disabled="false"
                            id="checkbox-7"
                            name="dataprivacy"
                            required=""
                            type="checkbox"
                            value="false"
                          />
                          
                        </div>
                      </div>
                      <!---->
                    </div>
                    
                  </div>
                  <!---->
                  <!---->
                </div>
                <span
                  class="ml-6 ml-sm-4 pt-2 newsletter-dataprivacy"
                  data-v-81cc7c1c=""
                >
                  $t('home.newsletterSection.newsletterForm.privacy') 
                  <a
                    class="v-btn v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain anchor-link anchor-link--text dataprivacy-link"
                    data-v-1925b644=""
                    data-v-81cc7c1c=""
                    href="/datenschutz"
                  >
                    
                    <span
                      class="v-btn__overlay"
                    />
                    <span
                      class="v-btn__underlay"
                    />
                    
                    <!---->
                    <span
                      class="v-btn__content"
                      data-no-activator=""
                    >
                      
                      
                      $t('home.newsletterSection.newsletterForm.privacyLinkLabel')
                      
                      
                    </span>
                    <!---->
                    <!---->
                  </a>
                </span>
              </div>
            </div>
            <div
              class="v-row align-content-center mt-0"
              data-v-81cc7c1c=""
            >
              <div
                class="v-col pt-0"
                data-v-81cc7c1c=""
              >
                <transition-group-stub
                  appear="false"
                  css="true"
                  data-v-81cc7c1c=""
                  persisted="false"
                >
                  <!--v-if-->
                  <!--v-if-->
                </transition-group-stub>
              </div>
            </div>
            
          </form>
        </div>
      </div>
    </div>
  </div>
  <!-- &lt;v-row class="mt-12" align="center"&gt;
      &lt;v-col cols="5"&gt;
        &lt;h3 class="section-subheadline"&gt;
          {{ $t('home.newsletterSection.currentNewsletter') }}
        &lt;/h3&gt;
      &lt;/v-col&gt;
      &lt;v-col cols="7"&gt;
        &lt;MainButton
          variant="download"
          size="auto"
          :label="$t('home.newsletterSection.newsletterDownloadBtn')"
          type="submit"
          download
          :href="newsletterDownloadPath"
        &gt;
        &lt;/MainButton&gt;
      &lt;/v-col&gt;
    &lt;/v-row&gt; -->
</div>
`;
